catalog/src/main/java/client/InventoryServiceClient.java

```diff
@@ -1,52 +1,31 @@
 package client;
 
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.List;
-
-import javax.json.JsonArray;
-import javax.json.JsonValue;
-import javax.ws.rs.client.Client;
-import javax.ws.rs.client.ClientBuilder;
-import javax.ws.rs.client.WebTarget;
+import javax.enterprise.context.Dependent;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.ProcessingException;
+import javax.ws.rs.Produces;
 import javax.ws.rs.core.MediaType;
+import java.util.List;
+import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
 
-import org.eclipse.microprofile.config.Config;
-import org.eclipse.microprofile.config.ConfigProvider;
-
-import com.google.gson.Gson;
-
-
-
-public class InventoryServiceClient {
-
-	  Client client = ClientBuilder.newClient();
-
-	  Config config = ConfigProvider.getConfig();
-	  
-	  String inv_url = config.getValue("inventory_url", String.class);
-	  String inv_health = config.getValue("inventory_health", String.class);
 
-	  public List<Item> getAllItems()
-	  {
-		List<Item> allItems = new ArrayList<>();
-		Client client = ClientBuilder.newClient();
-		WebTarget target = client.target(inv_url);
-		String s = target.request().get(String.class);
+@Dependent
+@RegisterRestClient
+@Path("/inv")
 
-		Gson gson = new Gson();
-		Item items[] = gson.fromJson(s, Item[].class);
+public interface InventoryServiceClient {
 
-		allItems = Arrays.asList(items);
+    // ProcessingException is temp until further error handling is implemented
 
-		return allItems;
-	  }
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("/inventory")
+    List<Item> getAllItems() throws ProcessingException;
 
-	  public String healthCheck(){
-		  Client client = ClientBuilder.newClient();
-		  WebTarget target = client.target(inv_health);
-		  String s = target.request().get(String.class);
-		  return s;
-	  }
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("/check")
+    String healthCheck() throws ProcessingException;
```

Comment:
We will replace this with MP Health eventually.

Reply:
Yeah I am working on it and replace the same once the mp health implementation is done. For now, I think David is using the default one we already have there. Thank you.